document/src/test/java/com/yahoo/document/json/JsonReaderTestCase.java

```diff
@@ -1400,23 +1404,27 @@ public void tensor_modify_update_treats_the_input_tensor_as_sparse() {
 
     @Test
     public void tensor_modify_update_on_non_tensor_field_throws() {
-        exception.expect(IllegalArgumentException.class);
-        exception.expectMessage("A modify update can only be applied to tensor fields. Field 'something' is of type 'string'");
-        JsonReader reader = createReader(inputJson("{ 'update': 'id:unittest:smoke::doc1',",
-                "  'fields': {",
-                "    'something': {",
-                "      'modify': {} }}}"));
-        reader.readSingleDocument(DocumentParser.SupportedOperation.UPDATE, "id:unittest:smoke::doc1");
+        try {
+            JsonReader reader = createReader(inputJson("{ 'update': 'id:unittest:smoke::doc1',",
+                                                       "  'fields': {",
+                                                       "    'something': {",
+                                                       "      'modify': {} }}}"));
+            reader.readSingleDocument(DocumentParser.SupportedOperation.UPDATE, "id:unittest:smoke::doc1");
```

Comment:
We should fail on the next line if the exception was not thrown.

Reply:
Good catch - thanks